tests/OTAnalytics/plugin_video_processing/test_video_reader.py

```diff
@@ -1,24 +1,58 @@
 from datetime import timedelta
 from pathlib import Path
 
-from OTAnalytics.domain.track import PilImage
-from OTAnalytics.plugin_video_processing.video_reader import OpenCvVideoReader
-
-
-class TestOpenCvVideoReader:
-    video_reader = OpenCvVideoReader()
+import av
+import pytest
+from PIL import Image, ImageChops
 
-    def test_get_image_possible(self, cyclist_video: Path) -> None:
-        image = self.video_reader.get_frame(cyclist_video, 1)
+from OTAnalytics.domain.track import PilImage
+from OTAnalytics.domain.video import VideoReader
+from OTAnalytics.plugin_video_processing.video_reader import (
+    PyAvVideoReader,
+    av_to_image,
+)
+
+
+class TestPyAVVideoReader:
+    def read_expected_frames(self, video_path: Path) -> list[Image.Image]:
+        expected_frames = []
+        with av.open(str(video_path.absolute())) as container:
+            for frame in container.decode(video=0):
+                expected_frames.append(av_to_image(frame).as_image())
+        return expected_frames
+
+    @pytest.fixture
+    def video_reader(self) -> VideoReader:
+        return PyAvVideoReader()
+
+    def test_get_image_possible(
+        self, cyclist_video: Path, video_reader: VideoReader
+    ) -> None:
+        image = video_reader.get_frame(cyclist_video, 1)
         assert isinstance(image, PilImage)
 
-    def test_get_frame_out_of_bounds(self, cyclist_video: Path) -> None:
-        image = self.video_reader.get_frame(cyclist_video, 100)
+    def test_get_frame_out_of_bounds(
+        self, cyclist_video: Path, video_reader: VideoReader
+    ) -> None:
+        image = video_reader.get_frame(cyclist_video, 100)
         assert isinstance(image, PilImage)
 
-    def test_get_frame_number_for(self, cyclist_video: Path) -> None:
+    def test_get_frame_number_for(
+        self, cyclist_video: Path, video_reader: VideoReader
+    ) -> None:
         delta = timedelta(seconds=1)
 
-        frame_number = self.video_reader.get_frame_number_for(cyclist_video, delta)
+        frame_number = video_reader.get_frame_number_for(cyclist_video, delta)
 
         assert frame_number == 20
+
+    @pytest.mark.parametrize("frame_num", (1, 5))
```

Comment:
Assuming the video has 60 frames, can we test the edge cases 0, 60, 61?

Reply:
I added the edge cases: -1, 0, 58, 59, 60 and 61